rust/src/core.rs

```diff
@@ -24,12 +24,14 @@ use crate::debug_validate_fail;
 /// Opaque C types.
 pub enum DetectEngineState {}
 pub enum AppLayerDecoderEvents {}
-pub enum AppLayerParserState {}
 
-// From app-layer-events.h
-pub type AppLayerEventType = std::os::raw::c_int;
-pub const APP_LAYER_EVENT_TYPE_TRANSACTION : i32 = 1;
-pub const APP_LAYER_EVENT_TYPE_PACKET      : i32 = 2;
+#[repr(C)]
```

Comment:
If this is used by C, should we use `#[repr(i32)]`?

Reply:
Just want to make sure there are no mismatched types anywhere.

Reply:
Currently, there is a mismatch (int in C, i32 in rust) , this PR fixes it